### ISSUE TITLE
Test Plan: v6-agoricNames upgrade

### ISSUE THREAD

**Issue by anilhelvaci** [Agoric/Agoric#10647]
## What is the Problem Being Solved?
We need to test `agoricNames` keep working after a upgrade in Emerynet and any other test networks that are required.

## Description of the Design
| Features to test |  `a3p` local net | Test Net (Emerynet, XNet..) | Form | Verify | Core-eval required |
|:----------:| :----------: | :---: | --- | :--- | :---: |
| Existing values survive upgrade | yes | ? | `agd` and core eval | * write to `[issuer, brand, vbankAssets, oracleBrand, installation, instance]` <br> * make sure old values are present in `vstorage` | yes |
| Send non-vbank asset | yes | ? | core eval | Receiving end gets the funs| yes | 
| Open a vault<br> (or a PSM swap) | yes | yes | `agd` | IST balance increases | no |
| Ephemeral `onUpdate` callbacks keep working | yes | ? | core eval | * write to child with ephemeral `onUpdate` <br> * observe in `vstorage` | yes 

### Why do we need core-evals?
We have two main criteria when testing the `agoricNames` upgrade:
1. Do the old values survive the upgrade?
   * The values should be available
   * If the value is an object, it should be alive and usable 
2. Can we write to anything new?
   * We should be able to;
     * write new data to an existing child
     * create a new child and write into it

Core-evals are the only way we can write to `agoricNames` as its nameAdmin,`agoricNamesAdmin`, is available in the bootstrap space.

## Security Considerations
Any misbehavior of `agoricNames` would cause damage on `smartWallet`, `interProtocol`  and probably some other very important parts of the system. So I believe we should consider this test as **critical**.

## Scaling Considerations
Not a major issue.

## Test Plan
```mermaid
flowchart LR
    A@{ shape: circle, label: "Upgrade 19 CoreEvals finish" } --> B(observe in the logs)
    B --> C(agoricNames upgrade successful) --> I
    A --> D{Running test core-evals?}
    D -- yes --> L(observe in the logs)
    L --> E(sendin non-vbank asset successful) --> I
    L --> K(writing to agoricNames successful) --> I
    L --> J(registering ephemeral onUpdate successful) --> I
    D -- no --> I
    A --> F(Open a vault)
    subgraph Inter-Protocol-Operation
    F --> G(wait until reserve receives fee)
    end
    subgraph Existing-Values-Survive
    f --> g(run the core-eval)
    g --> e(snapshot again)
    e --> a(compare)
    end
    subgraph Send-non-vbank
    b -- core eval sends the asset --> v(lookup published.wallet)
    v --> s(observe balance increase)
    end
    subgraph Epehemeral-onUpdate
    l --> n(core eval registers onUpdate)
    n --> m(upgrade agoricNames)
    m --> c(core eval trigger onUpdate)
    c --> x(observe change in vstorage)
    end
    Existing-Values-Survive --> I@{ shape: framed-circle, label: "Stop" }
    Inter-Protocol-Operation --> I
    Send-non-vbank --> I
    Epehemeral-onUpdate --> I
    D -- yes --> f(snapshot agoricNames)
    D -- yes --> b(run core-eval)
    D -- yes --> l(run prep core-eval)
```

## Upgrade Considerations
This is intended to be a workspace for laying out the plan for testing `agoricNames` upgrade which is planned to take place in Upgrade 19.

**Comment by anilhelvaci** [comment]
assigned you for review @Chris-Hibbert

**Comment by Chris-Hibbert** [comment]
ask @samsiegart if significant parts of the UIs depend on agoricNames. If so, then we can check that agoricNames is working  by exercising the UIs.

**Comment by anilhelvaci** [comment]
> ask @samsiegart if significant parts of the UIs depend on agoricNames. If so, then we can check that agoricNames is working by exercising the UIs.

I think for `Opening a vault` and `PSM swap` operations going over FE is an option. I'm not aware of any other FEs that depend on `agoricNames`. Any suggestions? @samsiegart

**Comment by samsiegart** [comment]
> > ask @samsiegart if significant parts of the UIs depend on agoricNames. If so, then we can check that agoricNames is working by exercising the UIs.
> 
> I think for `Opening a vault` and `PSM swap` operations going over FE is an option. I'm not aware of any other FEs that depend on `agoricNames`. Any suggestions? @samsiegart

Yes, also keplr queries agoricNames when rendering offers in the signer UI
